src/server/src/main/java/io/cassandrareaper/service/RepairRunService.java

```diff
@@ -185,7 +196,32 @@ public RepairRun registerRepairRun(
       LOG.error(errMsg);
       throw new ReaperException(errMsg);
     }
-    return segments;
+
+    // Compute replicas per DC for each segment
+    List<Segment> segmentsWithReplicas = Lists.newArrayList();
+    for (Segment segment:segments) {
+      segmentsWithReplicas.add(
+          Segment.builder()
+                 .withBaseRange(segment.getBaseRange())
+                 .withTokenRanges(segment.getTokenRanges())
+                 .withReplicas(getDCsByNodeForRepairSegment(targetCluster, segment, repairUnit.getKeyspaceName()))
+                 .build());
+    }
+
+    return segmentsWithReplicas;
+  }
+
+  private Map<String, String> getDCsByNodeForRepairSegment(
+      Cluster cluster,
+      Segment segment,
+      String keyspace) throws ReaperException {
+
+    JmxProxy jmxConnection = clusterFacade.connect(cluster);
+    // when hosts are coming up or going down, this method can throw an UndeclaredThrowableException
+    Collection<String> nodes = clusterFacade.tokenRangeToEndpoint(cluster, keyspace, segment);
```

Comment:
How do we deal with failure here? Previously we'd just move on to another segment? Now the whole repair is jammed/failed because initialisation fails?

Reply:
It would fail indeed.
I can add a retry mechanism to at least give it a few attempts before failing the repair init.
wdyt?

Reply:
Added a retry mechanism

Reply:
Is this retry mechanism at repair-run initialisation enough? 

What happens when there are range movements during an ongoing repair-run? How do we detect that? How does it fail? Why (should) it fail? Do the docs need to be updated to describe these types of situations?

Reply:
My take on this is that repair should fail if there's a range movement. Such operations will probably break the guarantees that repair brings.

Reply:
I made it so that repair fails in case of topology change with an appropriate error message.
Anyway, adding/removing nodes would probably lead to the same result as token ranges that were computed for the segments aren't valid anymore.

wdyt @michaelsembwever ?

Reply:
👍 

I agree that the whole repair run should fail (quickly and cleanly) on any token range movement.